CRM/Core/BAO/Block.php

```diff
@@ -79,28 +81,15 @@ public static function &getValues($blockName, $params) {
    * Given the list of params in the params array, fetch the object
    * and store the values in the values array
    *
-   * @param Object $block
-   *   Typically a Phone|Email|IM|OpenID object.
-   *
-   * @return array
-   *   Array of $block objects.
+   * @param Civi\Api4\Generic\DAOGetAction $request
+   *   An API4 Get request, typically of Phone|Email|IM|OpenID.
    */
-  public static function retrieveBlock($block) {
-    // we first get the primary location due to the order by clause
-    $block->orderBy('is_primary desc, id');
-    $block->find();
+  public static function retrieveBlock(Civi\Api4\Generic\DAOGetAction $request) {
+    $blocks = (array) $request->addOrderBy('is_primary', 'DESC')->addOrderBy('id')->execute();
```

Comment:
The 2 big differences with Api4 vs BAO are

1. Custom fields are named differently (not an issue here as I think custom data is fetched elsewhere)
2. Html is no longer escaped. Probably *is* an issue.